proposals/text/3875-table-span.md

```diff
@@ -0,0 +1,228 @@
+within the table since the table is either returned as a list of lists or a pandas dataframe in Haystack.
+This would allow users to easily look up the answer in the returned table to fetch the text directly from the table,
+identify the row or column labels for that answer, or generally perform operations on the table near or around the
+answer cell.
+
+# Basic Example
+
+When applicable, write a snippet of code showing how the new feature would be used.
+```python
+import pandas as pd
+from haystack.nodes import TableReader
+from haystack import Document
+
+data = {
+    "actors": ["brad pitt", "leonardo di caprio", "george clooney"],
+    "age": ["58", "47", "60"],
+    "number of movies": ["87", "53", "69"],
+    "date of birth": ["18 december 1963", "11 november 1974", "6 may 1961"],
+}
+table_doc = Document(content=pd.DataFrame(data), content_type="table")
+reader = TableReader(model_name_or_path="google/tapas-base-finetuned-wtq", max_seq_len=128)
+prediction = reader.predict(query="Who was in the most number of movies?", documents=[table_doc])
+answer = prediction["answers"][0]
+
+# New feature
+print(answer.context.iloc[answer.offsets_in_context[0].row, answer.offsets_in_context[0].col])
+```
+
+# Motivation
+
+## Why do we need this feature?
+To allow users to easily look up the answer cell in the returned table to fetch the answer text
+directly from the table, identify the row or column labels for that answer, or generally perform operations on the table
+near or around the answer cell.
+
+Currently, we return the location of the answer in the **linearized** version of the table, so we can use the
+`Span` dataclass. The `Span` dataclass is reproduced below:
+```python
+@dataclass
+class Span:
+    start: int
+    end: int
+    """
+    Defining a sequence of characters (Text span) or cells (Table span) via start and end index.
+    For extractive QA: Character where answer starts/ends
+    For TableQA: Cell where the answer starts/ends (counted from top left to bottom right of table)
+
+    :param start: Position where the span starts
+    :param end:  Position where the span ends
+    """
+```
+This is inconvenient for users because they would need to know how the table is linearized (column major or row major)
+so they could reconstruct the column and row indices of the answer before they could locate the answer cell in the table.
+
+## What use cases does it support?
+Some examples are already stated above but to recap, to easily perform operations on the table near or around the answer cell.
+
+## What's the expected outcome?
+The addition of a new dataclass called `TableSpan` that would look like
+```python
+@dataclass
+class TableSpan:
+    col: int
+    row: int
+    """
+    Defining a table cell via the column and row index.
+
+    :param col: Column index of the cell
+    :param row: Row index of the cell
+    """
```

Comment:
I imagine that the `TableCell` version will be more common because we will rarely return continuous cells of the table as one answer. And currently users expect that if multiple cells are returned that they would come as a list of `TableCell`s. 

So let’s go with the rename to `TableCell` and leave `row` and `col` as ints.